card-gallery/expressive-search.js

```diff
@@ -0,0 +1,968 @@
+    this.innatePowerName = null;
+
+    // String, nullable
+    // Only non-null for hero character cards
+    this.innatePowerEffect = null;
+
+    // String, nullable
+    // Only non-null for event cards
+    this.eventRuleTitle = null;
+
+    // String, nullable
+    // Only non-null for event cards
+    this.eventRuleEffect = null;
+
+    // String, nullable
+    this.setup = null;
+
+    // String, nullable
+    this.gameText = null;
+
+    // String, nullable
+    this.advancedGameText = null;
+
+    // String, nullable
+    this.featuredIssue = null;
+
+    // String, nullable
+    this.flavorText = null;
+
+    // String, nullable
+    this.flavorTextAttribution = null;
+
+    // String, nullable
+    this.incapCaption = null;
+
+    // String array, nullable
+    // An hypothetical card with an incap side and no incap options would be an empty array
+    this.incapOptions = null;
+
+    // String, nullable
+    this.incapFeaturedIssue = null;
+
+    // String, nullable
+    this.backDescription = null;
+    
+    // String, nullable
+    // A hypothetical card with a back side but not keywords on it would be an empty array
+    this.backKeywords = null;
+
+    // Integer, nullable
+    this.backHp = null;
+
+    // Array of strings, non-null
+    this.backNemesisIcons = [];
+
+    // String, nullable
+    this.backGameText = null;
+
+    // String, nullable
+    this.backAdvancedGameText = null; 
+
+    // String, nullable
+    this.collectionFlavorText = null;
+
+    // String, nullable
+    this.collectionFeaturedIssue = null;
+
+    // String, nullable
+    this.rewardATitle = null;
+
+    // String, nullable
+    this.rewardAFlavorText = null;
+
+    // String, nullable
+    this.rewardAGameText = null;
+
+    // String, nullable
+    this.rewardBTitle = null;
+
+    // String, nullable
+    this.rewardBFlavorText = null;
+
+    // String, nullable
+    this.rewardBGameText = null;
+
+    // String array, nullable
+    this.collectionIssues = null;
+
+
+    // ==========
+    // Metadata
+    // ==========
+    // Integer, nullable
+    // Technically character cards have a quantity of 1, but the only time anyone cares about this field is when it refers to cards in an actual deck
+    this.quantity = null;
+
+    // String, non-null
+    this.set = ""
+
+    // Integer, nullable
+    // Only base hero character cards get complexity. Variants of that hero have a null complexity, as do hero cards in the deck.
+    this.complexity = null;
+
+    // Integer, nullable
+    // Only primary base villain character cards get difficulty. Secondary villains and critical events have a null complexity, as do villain cards in the deck.
+    this.difficulty = null;
+
+    // String, non-null
+    // Enum of:
+    //  - hero
+    //  - villain
+    //  - environment
+    //  - event
+    //  - critical event
+    this.type = "";
+
+    // String, non-null
+    // Enum of:
+    //  - deck
+    //  - character
+    //  - event
+    this.kind = "";
```

Comment:
Kind and Type are kind of made up. We know hero, villain, & environment cards exist, but events need a type of some kind so I gave them their own type. Maybe their type should be villain? especially since the flipside of a CE is a villain?

But then I also want a way to search for events specifically, and to search for critical vs non-critical events (which I'm calling "standard events" in the code). This all seems a little fuzzy. I'm open to opinions here.